src/proof/arith_proof.cpp

```diff
@@ -24,15 +24,84 @@
 #include "proof/theory_proof.h"
 #include "theory/arith/constraint_forward.h"
 #include "theory/arith/theory_arith.h"
+#include "theory/rewriter.h"
 
 #define CVC4_ARITH_VAR_TERM_PREFIX "term."
 
 namespace CVC4 {
+using namespace CVC4::theory;
 
 inline static Node eqNode(TNode n1, TNode n2) {
   return NodeManager::currentNM()->mkNode(kind::EQUAL, n1, n2);
 }
 
+// Proofs are supported only for contraints of the form __ <= __ .
+inline static bool allChildrenGeq(const Node& conflict)
+{
+  bool result = true;
+  size_t nChildren = conflict.getNumChildren();
+  for (size_t i = 0; i < nChildren; i++)
+  {
+    const Node& n = conflict[i];
+    const Node& nonneg = n.getKind() == kind::NOT ? n[0] : n;
+    if (nonneg.getKind() == kind::GEQ)
+    {
+      continue;
+    }
+    else
+    {
+      result = false;
+      break;
+    }
+  }
+  return result;
+}
+
+// Verify that that the farkas coefficients indeed create a contradiction
+inline static bool hasContradiction(
```

Comment:
```suggestion
static bool hasContradiction(
```

(may have to reformat)

Reply:
fixed in next commit.